dirty_cat/similarity_encoder.py

```diff
@@ -316,11 +394,61 @@ def transform(self, X):
             out = np.empty((len(X), total_length), dtype=self.dtype)
             last = 0
             for j, cats in enumerate(self.categories_):
-                out[:, last:last + len(cats)] = ngram_similarity(X[:, j], cats,
-                                                                 ngram_range=(min_n, max_n),
-                                                                 hashing_dim=self.hashing_dim,
-                                                                 dtype=np.float32)
+                if fast:
+                    encoded_Xj = self._ngram_similarity_fast(X[:, j], j)
```

Comment:
Do we have any reason not to want "fast"? Given that we are already paying the price of fitting the vectorizers during fit.

Reply:
It allowed an easy comparison between the two methods during development. Indeed I think this can be removed before merging, but before that I can show the output a minimal script showing time comparisons for the two methods.